src/get-activity.ts

```diff
@@ -18,6 +18,15 @@ import {
   getPullRequestReviews,
 } from "./start";
 
+export enum CommentType {
+  REVIEW = 0b1,
+  ISSUE = 0b10,
+  ASSIGNEE = 0b100,
+  ISSUER = 0b1000,
+  COLLABORATOR = 0b10000,
+  CONTRIBUTOR = 0b100000,
+}
+
 export class GetActivity {
   constructor(private _issueParams: IssueParams) {}
   self: Promise<GitHubIssue> | GitHubIssue | null = null;
```

Comment:
I think we can remove this `Promise<>` because its awaited.

Also does `null` here indicate that the request to get issue/comments/reviews failed? If so in that case it'd be better if the whole calculation failed because we don't have full data.

Reply:
@pavlovcik should know better on this one

Reply:
My guess on the `null` is because the value cannot be assigned on the constructor due to the `async` nature of the call to retrieve it

Reply:
I guess my aspiration would be that we can handle situations where we fail to retrieve the data elegantly. What about an exponential back off? The GitHub action has the ability to run for six hours!

Otherwise, if that's a bad idea then yes I agree with whilefoo that it should throw and exit the run. 

I'm still definitely leaning towards exponential backoff idea though, which implies that it should never be null, but instead perhaps, undefined.

Reply:
I removed all the `Promise` and only this one can remain `null` on a fail retrieval attempt. Since you created another task for it I will resolve this conversation.